packages/testing/src/execution_testing/forks/forks/forks.py

```
@@ -2309,6 +2354,59 @@ def blob_base_cost(
         del block_number, timestamp
         return 2**13  # EIP-7918 new parameter
 
+    @classmethod
+    def gas_costs(
+        cls, *, block_number: int = 0, timestamp: int = 0
+    ) -> GasCosts:
+        """Return the gas costs for the fork."""
+        from ethereum.forks.osaka.vm import gas as g
+        from dataclasses import fields
+
+        kwargs = {}
+        for field in fields(GasCosts):
+            if hasattr(g, field.name):
+                kwargs[field.name] = int(getattr(g, field.name))
+
+        return GasCosts(**kwargs)
```

Should we return an error here if the field is not found rather than skipping it silently? Not sure if this is intended.

---

Yes, we should return an error here like [this](https://github.com/ethereum/execution-specs/pull/1776#discussion_r2576239101)